reject :talk ++command pokes from foreign ships

joemfb opened the pull request:
closes #208 

Crashing rejects the poke, right? (As opposed to producing `[~ +>.$]`, which would accept and acknowledge?)

juped commented:
Yeah, I think that's right. But @cgyarvin needs to provide us with a gall design sketch.

joemfb commented:
And I initially botched that example ... for the benefit of emailers:

```
?|  =(our.hid src.hid)
    &(?=($earl (clan src.hid)) =(our.hid (sein src.hid)))
==
```

chc4 commented:
It probably should allow pokes from your moons. :dojo for example allows it, along with peering via sole. https://github.com/urbit/arvo/blob/master/app/dojo.hoon#L1102

There should probably be a helper function in zuse for this check, really, since it's what you usually want.

joemfb commented:
I added `++team` to `zuse` - seems like an appropriate name ...

To see if the crash "rejects" the poke, I added a `++coup` arm to a test app on a comet and poked `:talk` on a different comet. It was called, with a `++tang` showing the trace:

```
mack
/~tipsyt-middep-nommep-simdel--dapmyl-lapdev-bonnes-falfyr/base/0/app/talk:<[2.166 3].[2.170 24]>
/~tipsyt-middep-nommep-simdel--dapmyl-lapdev-bonnes-falfyr/base/0/app/talk:<[2.165 3].[2.170 24]>
```

ohAitch commented:
Naively, I would expect this to break [ra-transmit](https://github.com/urbit/arvo/blob/ab73f36/app/talk.hoon#L1568-L1571), i.e. all message posting. Does this work with like, a channel on a fakezod that a fakenec joins and writes to?

joemfb commented:
Sorry about that -- I opened the PR and then went mostly offline on vacation ... I saw it was merged, so I assumed it was fine
